src/tools/w3dviewer/mainfrm.cpp

```diff
@@ -597,7 +598,42 @@ void CMainFrame::OnProperties()
 
 void CMainFrame::OnGenerateLOD()
 {
-    // do later
+    CDataTreeView *view = static_cast<CDataTreeView *>(m_splitter.GetPane(0, 0));
+
+    if (view == nullptr || view->GetSelectedItemName() == nullptr) {
+        return;
+    }
+
+    CW3DViewDoc *doc = static_cast<CW3DViewDoc *>(GetActiveDocument());
+
+    if (doc != nullptr) {
+        const char *name = view->GetSelectedItemName();
+
+        int len = strlen(name);
+        char c = name[len - 2];
+        char c2 = name[len - 1];
+        bool format = false;
+
+        if (((c == 'L' || c == 'l') && c2 >= '0' && c2 <= '9')
+            || (format = true,
```

Comment:
format true is only set if the c2 conditions below are met

Reply:
also isn't this isalpha?

Reply:
isalpha can't be inlined.

Reply:
Fixed.